jenkinsapi/jenkins.py

```diff
@@ -646,7 +649,14 @@ def plugins(self):
 
     def get_plugins(self, depth: int = 1) -> Plugins:
         url = self.get_plugins_url(depth=depth)
-        return Plugins(url, self)
+        # If the plugins object is not already created or the baseurl has changed
+        # the we recreate a new one
+        if (
+            not hasattr(self, "_get_plugins")
+            or self._get_plugins.baseurl != url
+        ):
+            self._get_plugins = Plugins(url, self)
```

Comment:
I'd prefer something like `self._plugin_cache` to make it clear that it is data and not a method.